platform-ajax/src/main/typescript/com/softicar/platform/ajax/engine/KeepAlive.ts

```diff
@@ -1,52 +1,57 @@
+
 /**
  * This class takes care that no session timeout occurs
  * by sending a keep-alive message to the server.
  * <p>
  * @param delay the number of milliseconds between every 
  *              keep-alive message sent to the server
  */
-function KeepAlive(delay)
-{
+class KeepAlive {
+	private delay: number;
+	private handlerId: number;
+
+	constructor(delay: number) {
+		this.delay = delay;
+		this.handlerId = -1;
+	}
+
 	/**
 	 * This will schedule a new keep-alive message.
 	 * <p>
 	 * Any keep-alive message that was scheduled before the
 	 * call to this function will be canceled.
 	 */
-	this.schedule = function()
-	{
+	schedule() {
 		// clear any previously scheduled timeout
-		if(m_handler != null)
-			clearTimeout(m_handler);
+		if(this.handlerId >= 0) {
+			clearTimeout(this.handlerId);
+		}
 
 		// schedule new timeout with default delay		
-		m_handler = setTimeout(m_handleKeepAliveTimeout, m_delay);
-	};
+		this.handlerId = setTimeout(this.handleKeepAliveTimeout, this.delay);
+	}
 
-	var m_handleKeepAliveTimeout = function()
-	{
+	handleKeepAliveTimeout() {
 		// ignore if the session timed out
-		if(SESSION_TIMED_OUT)
+		if(SESSION_TIMED_OUT) {
 			return;
+		}
 	
 		// try to get the global event lock
-		if(lock(LOCK_REASON_KEEP_ALIVE))
-		{
-			var parameters = 
-			{
+		if(lock(LOCK_REASON_KEEP_ALIVE)) {
+			let parameters = {
 				'a': AJAX_REQUEST_KEEP_ALIVE
 			};
-			
-			AQ_enqueueAction(new AQ_ServerRequestAction(parameters));
-			AQ_executeNextAction();
-		}
-		else
+
+			ACTION_QUEUE.enqueueAction(new ServerRequestAction(parameters, null));
+			ACTION_QUEUE.executeNextAction();
+		} else {
 			// Since the global event lock was active there was no point
 			// in sending a keep-alive, anyway. So, just reschedule a new
 			// timeout.
-			GLOBAL.scheduleKeepAlive();
-	};
+			KEEP_ALIVE.schedule();
+		}
+	}
+}
 
-	var m_delay = delay;
-	var m_handler = null;
-};
+let KEEP_ALIVE = new KeepAlive(3*60*1000);
```

Comment:
We should probably extract a constant here.

Reply:
I can do that.

Reply:
did so